### ISSUE TITLE
Regarding the display of media file names

### ISSUE THREAD

**Issue by fengfrw** [jhe2/gmu#8]
Hope to support Chinese display of media file names, thank you!

**Comment by jhe2** [comment]
For chinese characters to be supported, I need to implement True Type font rendering in Gmu. Currently Gmu renders the characters on display via bitmap fonts to get that retro look. Implementing True Type font rendering is something I would like to do and with that, Chinese would be supported as well (given a font is used that includes the required glyphes). It might take a while until I get around to implementing that, though. I'll let you know once I do!

**Comment by fengfrw** [comment]
thank you very much! Looking forward to your completion.

**Comment by LucasGGamerM** [comment]
Yeah, this is a thing I am looking forward to. Not due to Chinese, but mainly for Japanese songs, which sadly appear with blank names.

**Comment by ghost** [comment]
Could it, in the meantime, include bitmap fonts that support such languages?  Although at this font size it would be difficult to read, at least there wouldn't be large blank areas.

If you implement TrueType font rendering, but still want "the retro look," you might want to turn off antialiasing and hinting.  Though it is necessary to ensure a good font is used if so.  (Some are _horrible_ without AA!)  Noto Sans has produced pretty good results down to some quite small sizes for me (it seems to be good down to size 7 I think.)  I'm not sure if the licensing is compatible, but if so it may be a good option.

**Comment by jhe2** [comment]
Using bitmap fonts for languages with so many characters isn't very practical unfortunately, but I have been working on true type font rendering in the meantime. You can check out the current state of things if you like. I've added basic TTF rendering support with commit a7697919deefc3e5c8706d8d7821d688b31ede19 and a theme using true type fonts in another commit. There are still a few thing left to do, but that shouldn't take me very long.
Funny that you mention Noto Sans, that's exactly the font I've been using for the theme that I've included with Gmu. :-)
